sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/MetricFeedback/MetricFeedback.cs

```diff
@@ -43,5 +44,31 @@ internal MetricFeedback(string metricId, FeedbackDimensionFilter dimensionFilter
 
         /// <summary> The dimension filter. </summary>
         public FeedbackDimensionFilter DimensionFilter { get; internal set; }
+
+        internal static MetricFeedback DeserializeMetricFeedback(JsonElement element)
+        {
+            if (element.TryGetProperty("feedbackType", out JsonElement discriminator))
+            {
+                var discriminatorString = discriminator.GetString();
+
+                switch (discriminatorString)
+                {
+                    case "Anomaly":
+                        return MetricAnomalyFeedback.DeserializeMetricAnomalyFeedback(element);
+                    case "ChangePoint":
+                        return MetricChangePointFeedback.DeserializeMetricChangePointFeedback(element);
+                    case "Comment":
+                        return MetricCommentFeedback.DeserializeMetricCommentFeedback(element);
+                    case "Period":
+                        return MetricPeriodFeedback.DeserializeMetricPeriodFeedback(element);
+                    default:
+                        throw new ArgumentException($"Unknown feedback type returned by the service: {discriminatorString}");
+                }
+            }
+            else
+            {
+                throw new ArgumentException("The feedback type was not returned by the service");
+            }
```

Comment:
We should never reach these exceptions if we assume the service will behave as expected. Needed to add these code paths anyway to be able to compile the code.